src/Couchbase.Lite.Shared/API/Sync/Replicator.cs

```diff
@@ -265,10 +266,97 @@ public void Stop()
             });
         }
 
+        /// <summary>
+        /// Gets a list of document IDs that are going to be pushed, but have not been pushed yet
+        /// <item type="bullet">
+        /// <description>API is a snapshot and results may change between the time the call was made and the time</description>
+        /// </item>
+        /// </summary>
+        /// <returns>An immutable set of strings, each of which is a document ID</returns>
+        /// <exception cref="CouchbaseLiteException">Thrown if no push replication</exception>
+        /// <exception cref="CouchbaseException">Thrown if an error condition is returned from LiteCore</exception>
+        [NotNull]
+        public IImmutableSet<string> GetPendingDocumentIDs()
+        {
+            if (!IsPushing()) {
+                WriteLog.To.Database.E(Tag, CouchbaseLiteErrorMessage.PullOnlyPendingDocIDs);
```

Comment:
There is a helper function to both log and throw an exception that you might want to use here.

Reply:
Apply `LogAndThrow` everywhere that you can.